### PR TITLE
update container_client and some tests

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: azure-storage-blob/tests/blob/test_container.py [hunk 1]
```diff
@@ -404,8 +401,8 @@ def test_get_container_acl(self):
 
         # Assert
         self.assertIsNotNone(acl)
-        self.assertEqual(len(acl), 0)
-        self.assertIsNone(acl.public_access)
+        self.assertIsNone(acl.get('public_access'))
+        self.assertIsNone(acl.get('deserialized'))
```
Comment (annatisch) [inline azure-storage-blob/tests/blob/test_container.py]: Should this be `acl.get('signed_identifiers')`?

File: azure-storage-blob/tests/blob/test_container.py [hunk 2]
```diff
@@ -474,45 +470,61 @@ def test_set_container_acl_with_empty_signed_identifiers(self):
...
+    def test_set_container_acl_with_empty_access_policy(self):
         # Arrange
         container = self._create_container()
+        identifier = SignedIdentifier(id='empty', access_policy=None)
+        identifiers = [identifier]
+        container.set_container_acl(identifiers)
 
         # Act
-        container.set_container_acl()
+        container.set_container_acl(identifiers)
```
Comment (annatisch) [inline azure-storage-blob/tests/blob/test_container.py]: It seems like this line is called twice? (Also on 478)

Comment (xiafu-msft) [reply azure-storage-blob/tests/blob/test_container.py]: Thanks for catching this! It has been fixed in the next commit

File: azure-storage-blob/tests/blob/test_container.py (hunk 1 above)
Comment (xiafu-msft) [reply azure-storage-blob/tests/blob/test_container.py]: Yes, it's a typo, thanks for catching this. This is fixed in the second commit.

**Pull request merged**